### PR TITLE
Use Kotlin `vararg` to combine solo and batch APIs in `FhirEngine` and `Database`

### LINKED ISSUE #404: Combine the single and batch insertion APIs in FHIR engine and db
**Describe the Issue**
We have `FhirEngine.save`, `FhirEngine.saveAll`, `Database.insert`, `Database.insertAll`, `Database.insertRemote` and `Database.insertAllRemote`. They are redundant.

**Would you like to work on the issue?**
Use the Kotlin `varargs` keyword to combine them.

### PR DESCRIPTION
**Pull request by jingtang10**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straighforward changes).**

Fixes #404 

**Description**
Use Kotlin `vararg` to combine solo and batch APIs in `FhirEngine` and `Database`.

**Alternative(s) considered**
NA

**Type**
Code health

**Checklist**
- [x] I have read and acknowledge the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

File: core/src/main/java/com/google/android/fhir/FhirEngine.kt [hunk 1]
```diff
@@ -24,21 +24,13 @@ import org.hl7.fhir.r4.model.Resource
...
-  /**
-   * Saves a FHIR `resource` in the local storage. If the resource already exists, it will be
-   * overwritten
-   *
-   * @param <R> The resource type which should be a subtype of [Resource].
-   */
-  suspend fun <R : Resource> save(resource: R)
-
   /**
    * Saves a list of FHIR `resource` in the local storage. If any of the resources already exist,
```
Comment (deepankarb) [inline core/src/main/java/com/google/android/fhir/FhirEngine.kt:28]:
```suggestion
   * Saves one or more FHIR `resource` in the local storage. If any of the resources already exist,

**Review by deepankarb — APPROVED** [review]
Awesome! Minor suggestion on documentation.

**Comment by jingtang10** [comment]
Thanks @deepankarb! Sorry I had auto merge on. Will I corporate the proposed change in a follow-up.

File: core/src/main/java/com/google/android/fhir/FhirEngine.kt (hunk 1 above)
Comment (jingtang10) [reply core/src/main/java/com/google/android/fhir/FhirEngine.kt:28]: this is now being done in #395

**Pull request merged**